tooling/linux_repro_build_compare.sh

```diff
@@ -0,0 +1,146 @@
+    echo "Retrieving boot JDK $BOOTJDK_VERSION" && mkdir -p /usr/lib/jvm && curl -L "https://api.adoptopenjdk.net/v3/binary/version/jdk-${BOOTJDK_VERSION}/linux/${NATIVE_API_ARCH}/jdk/hotspot/normal/adoptopenjdk?project=jdk" | (cd /usr/lib/jvm && tar xpzf -)
+  fi
+  if [ ! -r "${LOCALGCCDIR}/bin/g++-${GCCVERSION}" ]; then
+    echo "Retrieving gcc $GCCVERSION" && curl "https://ci.adoptium.net/userContent/gcc/gcc$(echo "$GCCVERSION" | tr -d .).$(uname -m).tar.xz" | (cd /usr/local && tar xJpf -) || exit 1
+  fi
+  if [ ! -r temurin-build ]; then
+    git clone https://github.com/adoptium/temurin-build || exit 1
+  fi
+  (cd temurin-build && git checkout "$TEMURIN_BUILD_SHA")
+}
+
+checkAllVariablesSet() {
+    [ -z "$SBOM" ] || [ -z "${BOOTJDK_VERSION}" ] || [ -z "${TEMURIN_BUILD_SHA}" ] || [ -z "${TEMURIN_BUILD_ARGS}" ] || [ -z "${TEMURIN_VERSION}" ] && echo "Could not determine one of the variables - run with sh -x to diagnose" && sleep 10 && exit 1
+}
+
+installPrereqs
+downloadAnt
+
+echo "Retrieving and parsing SBOM from $SBOM_URL"
+curl -LO "$SBOM_URL"
+SBOM=$(basename "$SBOM_URL")
+BOOTJDK_VERSION=$(grep configure_arguments "$SBOM" | tr ' ' \\n | grep ^Temurin- | uniq | cut -d- -f2)
+GCCVERSION=$(tr ' ' \\n < "$SBOM" | grep CC= | cut -d- -f2 | cut -d\\ -f1)
+LOCALGCCDIR=/usr/local/gcc$(echo "$GCCVERSION" | cut -d. -f1)
+TEMURIN_BUILD_SHA=$(awk -F'"' '/buildRef/{print$4}' "$SBOM"  | cut -d/ -f7)
+TEMURIN_BUILD_ARGS=$(grep makejdk_any_platform_args "$SBOM" | cut -d\" -f4 | sed -e "s/--disable-warnings-as-errors --enable-dtrace --without-version-pre --without-version-opt/'--disable-warnings-as-errors --enable-dtrace --without-version-pre --without-version-opt'/" -e "s/ --disable-warnings-as-errors --enable-dtrace/ '--disable-warnings-as-errors --enable-dtrace'/" -e 's/\\n//g' -e "s,--jdk-boot-dir [^ ]*,--jdk-boot-dir /usr/lib/jvm/jdk-$BOOTJDK_VERSION,g")
+
+getTemurinVersion
+
+NATIVE_API_ARCH=$(uname -m)
+if [ "${NATIVE_API_ARCH}" = "x86_64" ]; then NATIVE_API_ARCH=x64; fi
+if [ "${NATIVE_API_ARCH}" = "armv7l" ]; then NATIVE_API_ARCH=arm; fi
+
+checkAllVariablesSet
+
+downloadTooling
+setEnvironment
+
+if [ ! -d "jdk-${TEMURIN_VERSION}" ]; then
+   if [ -z "$TARBALL_URL" ]; then
+       TARBALL_URL="https://api.adoptopenjdk.net/v3/binary/version/jdk-${TEMURIN_VERSION}/linux/${NATIVE_API_ARCH}/jdk/hotspot/normal/adoptopenjdk?project=jdk"
+   fi
+   echo Retrieving original tarball from adoptium.net && curl -L "$TARBALL_URL" | tar xpfz - && ls -lart "$PWD/jdk-${TEMURIN_VERSION}" || exit 1
+fi
+
+echo "  cd temurin-build && ./makejdk-any-platform.sh $TEMURIN_BUILD_ARGS 2>&1 | tee build.$$.log" | sh
+
+echo Comparing ...
+mkdir compare.$$
+tar xpfz temurin-build/workspace/target/OpenJDK*-jdk_*tar.gz -C compare.$$
+
+cleanBuildInfo
+
+diff -r "jdk-${TEMURIN_VERSION}" "compare.$$/jdk-$TEMURIN_VERSION" 2>&1 | tee "reprotest.$(uname).$TEMURIN_VERSION.diff"
+if [ "$(wc -l < "reprotest.$(uname).$TEMURIN_VERSION.diff")" -gt 0 ]; then
```

Comment:
Could we just use the return code from the diff command `$?` for this?

Reply:
The diff includes a pipe to "tee", so $? is always 0 for a successful "tee"

Reply:
True, although that could be resolved by just redirecting the output, checking the return code, and then `cat`ing the file, whch would make the code a bit simpler IMHO.
Take or leave that suggestion as you wish :-)

Reply:
Thanks, good idea, i've updated